notebooks/metatranscriptomics_pipeline_iSeqnorm.ipynb

```diff
@@ -0,0 +1,1477 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dnase_dna_concs_fp = './test_data/Quant/MiniPico/2024_NPH_Plates_3-6_Post_DNase_gDNA_Quant.txt'\n",
+    "\n",
+    "if not os.path.isfile(sample_dna_concs_fp):\n",
+    "    print(\"Problem! %s is not a path to a valid file\" % file)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Read in the DNA concentration output file**. It should be a .txt file and should look something like this:\n",
+    "    \n",
+    "```\n",
+    "##BLOCKS= 1\r\n",
+    "Group: Unknowns\r\n",
+    "Sample\tWells\tRFU_Values\tConcentration\tMean_Conc\tSD\tCV\tDilution\tAdjConc\t\r\n",
+    "01\tA1\t16352.000\t0.434\t0.434\t0.000\t0.0\t\t\t\r\n",
+    "02\tC1\t26061758.000\t37.588\t37.588\t0.000\t0.0\t\t\t\r\n",
+    "03\tE1\t1483519.000\t2.527\t2.527\t0.000\t0.0\t\t\t\r\n",
+    "04\tG1\t3015184.000\t4.712\t4.712\t0.000\t0.0\t\t\t\r\n",
+    "05\tI1\t16574.000\t0.434\t0.434\t0.000\t0.0\t\t\t\r\n",
+    "06\tK1\t14988.000\t0.432\t0.432\t0.000\t0.0\t\t\t\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dnase_dna_concs = read_pico_csv(dnase_dna_concs_fp, plate_reader='SpectraMax_i3x',conc_col_name='Post DNAse DNA Concentration')\n",
+    "\n",
+    "plate_df = pd.merge(plate_df, dnase_dna_concs, on='Well')\n",
+    "\n",
+    "well_col = 'Well'\n",
+    "\n",
+    "plate_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Enter the correct path to the Pico RNA concentration output**. This should be a txt-formatted file produced by the MiniPico assay on the condensed, 384-well plate. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dnase_rna_concs_fp = './test_data/Quant/MiniPico/2024_NPH_Plates_3-6_Post_DNase_RNA_Quant.txt'\n",
+    "\n",
+    "if not os.path.isfile(dnase_rna_concs_fp):\n",
+    "    print(\"Problem! %s is not a path to a valid file\" % file)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "dnase_rna_concs = read_pico_csv(dnase_rna_concs_fp, plate_reader='SpectraMax_i3x',conc_col_name='Post DNAse RNA Concentration')\n",
+    "\n",
+    "plate_df = pd.merge(plate_df, dnase_rna_concs, on='Well')\n",
+    "\n",
+    "well_col = 'Well'\n",
+    "\n",
+    "plate_df.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Visualize plate DNA/RNA concentrations after DNAse treatment:**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# get DNA & RNA post DNAse information\n",
+    "dna_dnase = make_2D_array(plate_df, data_col='Post DNAse DNA Concentration', well_col=well_col).astype(float)\n",
```

Comment:
Obviously, one can get away with not defining well_col in this cell because it has been defined above.  However, I vote for explicitly redefining it here ... it makes debugging difficult when one has to search through the whole notebook to find the last most recent re-definition of a variable.